### PR TITLE
Streaming.take(n).toList shouldn't blow up when n+1 is an error

### PR DESCRIPTION
**Pull request by ceedubs**
The `Cons(a, lt.map(_.take(n - 1)))` seems reasonable enough, but since `toList` calls `lt.value`, this blows up when `lt` is a deferred error.

### CODE REVIEW

File: core/src/main/scala/cats/data/Streaming.scala [hunk 1]
```diff
@@ -473,7 +473,9 @@ sealed abstract class Streaming[A] { lhs =>
     if (n <= 0) Empty() else this match {
       case Empty() => Empty()
       case Wait(lt) => Wait(lt.map(_.take(n)))
-      case Cons(a, lt) => Cons(a, lt.map(_.take(n - 1)))
+      case Cons(a, lt) =>
+        if (n == 1) Cons(a, Now(Empty()))
+        else Cons(a, lt.map(_.take(n - 1)))
```
Comment (non) [inline core/src/main/scala/cats/data/Streaming.scala]: Maybe I'm too fussy, but could we rewrite that line?

```scala
Cons(a, if (n == 1) Now(Empty()) else lt.map(_.take(n - 1))
```

**Comment by non** [comment]
:+1: (with-or-without my fussy suggestion honestly)

**Comment by non** [comment]
:+1:

**Comment by adelbertc** [comment]
Merge conflicts. Also one nitpicky comment :-)

File: core/src/main/scala/cats/data/Streaming.scala [hunk 2]
```diff
@@ -473,7 +473,8 @@ sealed abstract class Streaming[A] { lhs =>
     if (n <= 0) Empty() else this match {
       case Empty() => Empty()
       case Wait(lt) => Wait(lt.map(_.take(n)))
-      case Cons(a, lt) => Cons(a, lt.map(_.take(n - 1)))
+      case Cons(a, lt) =>
+        Cons(a, if (n ==1) Now(Empty()) else lt.map(_.take(n - 1)))
```
Comment (adelbertc) [inline core/src/main/scala/cats/data/Streaming.scala]: Eeeek, space between `==` and `1` :-)

**Comment by ceedubs** [comment]
@adelbertc thanks. Hopefully this should pass inspection now :)

**Comment by non** [comment]
:+1:

**Comment by adelbertc** [comment]
:+1:

**Pull request merged**